lib/usdUfe/ufe/UsdAttributeHolder.cpp

```diff
@@ -59,17 +61,58 @@ bool setUsdAttrMetadata(
 
     UsdUfe::AttributeEditRouterContext ctx(attr.GetPrim(), attr.GetName());
 
+    // These metadata have a USD equivalent. Use it to stay compatible with other DCCs:
+    static const auto knownUSDMetadata
+        = std::unordered_set<std::string> { UsdUfe::MetadataTokens->UIDoc,
+                                            UsdUfe::MetadataTokens->UIEnumLabels,
+                                            UsdUfe::MetadataTokens->UIName,
+                                            UsdUfe::MetadataTokens->UIFolder };
+
     PXR_NS::TfToken tok(key);
     if (PXR_NS::UsdShadeNodeGraph(attr.GetPrim())) {
         if (PXR_NS::UsdShadeInput::IsInput(attr)) {
             PXR_NS::UsdShadeInput input(attr);
-            input.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                input.ClearSdrMetadataByKey(tok);
+            } else {
+                input.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
         } else if (PXR_NS::UsdShadeOutput::IsOutput(attr)) {
             PXR_NS::UsdShadeOutput output(attr);
-            output.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                output.ClearSdrMetadataByKey(tok);
+            } else {
+                output.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
+        }
+    }
+
+    if (key == UsdUfe::MetadataTokens->UIDoc) {
+        attr.SetDocumentation(value.get<std::string>());
+        return true;
+    } else if (key == UsdUfe::MetadataTokens->UIEnumLabels) {
+        auto       sdfAttr = TfStatic_cast<SdfAttributeSpecHandle>(attr.GetPropertyStack().front());
+        const auto enumStrings = UsdUfe::splitString(value.get<std::string>(), ", ");
+        VtTokenArray allowedTokens;
+        allowedTokens.reserve(enumStrings.size());
+        for (const auto& tokenString : enumStrings) {
+            allowedTokens.push_back(TfToken(tokenString));
         }
+        sdfAttr->SetAllowedTokens(allowedTokens);
+        return true;
+    } else if (key == UsdUfe::MetadataTokens->UIFolder) {
+        // Translate '|' to ':'.
```

Comment:
Similar question I guess. I suppose we have no choice but "know" what the format is. Is there docs about this?